core/transaction_pool.go

```diff
@@ -180,30 +188,29 @@ func (pool *TxPool) SetLocal(tx *types.Transaction) {
 // validateTx checks whether a transaction is valid according
 // to the consensus rules.
 func (pool *TxPool) validateTx(tx *types.Transaction) error {
-	// Validate sender
-	var (
-		from common.Address
-		err  error
-	)
-
 	local := pool.localTx.contains(tx.Hash())
 	// Drop transactions under our own minimal accepted gas price
 	if !local && pool.minGasPrice.Cmp(tx.GasPrice()) > 0 {
 		return ErrCheap
 	}
 
-	// Validate the transaction sender and it's sig. Throw
-	// if the from fields is invalid.
-	if from, err = tx.From(); err != nil {
+	currentState, err := pool.currentState()
+	if err != nil {
+		return err
+	}
+
+	var from common.Address
+	if pool.homestead {
+		from, err = tx.From()
+	} else {
+		from, err = tx.FromFrontier()
+	}
+	if err != nil {
```

Comment:
There's actually a catch to this as we discussed already on Gitter is that you can kind of fuck the pool over and DOS it. I also believe you mean `FromFrontier`.

And yes you are right, this used to be `FromFrontier` and an additional check was done in the worker (and this is the second catch), you need additional complex code there to facilitate for this ... annoyance.